### PR TITLE
Build Errors using raspios full and lite

### PR DESCRIPTION
**Pull request by pavlic244**
Using raspios-lite or raspios-full imagebakery fails with different errors, either a package
cannot be removed, soft links in the image cannot be set or no space left on device.
Another problem is that sfdisk is still buggy in Buster so it has been replaced by parted
to resize rootfs-partition and image respectively.

Signed-off-by: Frank Pavlic <f.pavlic@kunbus.com>

### CODE REVIEW

File: customize_image.sh [hunk 1]
```diff
@@ -82,7 +106,7 @@ sed -r -i -e 's/#hdmi_force_hotplug=1/hdmi_force_hotplug=1/' \
 	  $IMAGEDIR/boot/config.txt
 
 # limit disk space occupied by logs
-ln -s ../cron.daily/logrotate $IMAGEDIR/etc/cron.hourly
+[ -f ../$IMAGEDIR/etc/cron.hourly/logrotate ] || ln -s ../cron.daily/logrotate $IMAGEDIR/etc/cron.hourly
```
Comment (zhan-kunbus) [inline customize_image.sh:109]: not quite sure this should be checked.
as I know, with this check  the repeatly customize an image might be works, but the result will not work. 
This checking might delay the finding of the error.

File: customize_image.sh [hunk 2]
```diff
@@ -60,6 +65,25 @@ cleanup() {
...
 trap cleanup ERR SIGINT
 
+# resize partition 2 to END=3964MB if necessary
+imgsize=$(parted "$1" unit b print | grep -e "\.img" | awk -F ":" '{gsub(/^[ \t]+|[B \t]+$/,"",$2); print $2}')
+[ "x$imgsize" = "x" ] && echo 1>&1 "Error: Image size not found" && exit 1
+secsize=$(parted "$1" unit b print | grep -e "Sector size" | awk -F "/" '{gsub(/^[ \t]+|[B \t]+$/,"",$3); print $3}')
+[ "x$secsize" = "x" ] && echo 1>&1 "Error: Sector size not found" && exit 1
+
+if [ $imgsize -lt 3900000000 ] ; then
+   disksize=3963617280
```
[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply customize_image.sh:75]: would it be fine if later on a bigger img is used?

File: customize_image.sh (hunk 1 above)
Comment (pavlic244) [reply customize_image.sh:109]: Without checking the build with raspios-full image will fail as this is file is already existing .

File: customize_image.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (pavlic244) [reply customize_image.sh:75]: There is already a comment above - # resize partition 2 to END=3964MB if necessary.

[the remark this answers is not in the document]
Comment (iluminat23) [reply customize_image.sh:75]: The comment is a bit vague. It is unclear why this limit exists. If someone new is reading the code it seems like a magical value.

File: customize_image.sh (hunk 1 above)
Comment (iluminat23) [reply customize_image.sh:109]: Ok, I think we need a ticket for this. Why does the file exist? Is it the identical link or a different?

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: customize_image.sh [hunk 3]
```diff
@@ -82,7 +106,7 @@ sed -r -i -e 's/#hdmi_force_hotplug=1/hdmi_force_hotplug=1/' \
 	  $IMAGEDIR/boot/config.txt
 
 # limit disk space occupied by logs
-ln -s ../cron.daily/logrotate $IMAGEDIR/etc/cron.hourly
```
[the remark this answers is not in the document]
Comment (pavlic244) [reply customize_image.sh:85]: I am not sure if this is a good idea - Reason for one commit is to get imagebakery fixed and working with all raspios-images. Moving this fix to an own commit would mean that using raspios-full imagebakery will fail.

[the remark this answers is not in the document]
Comment (iluminat23) [reply customize_image.sh:85]: Why? If you first make your changes to the symlink and then your other stuff on top of it.

[the remark this answers is not in the document]
Comment (pavlic244) [reply customize_image.sh:85]: good point :-)  I've thought the other way around

[the remark this answers is not in the document]
Comment (pavlic244) [reply customize_image.sh:85]: Well this one can be skipped - this was my fault . I've used an already processed image as base , in all three raspios-images the logrotate does not exist in cron.hourly.

File: customize_image.sh [hunk 4]
```diff
@@ -60,6 +65,25 @@ cleanup() {
 
 trap cleanup ERR SIGINT
 
+# resize partition 2 to END=3964MB if necessary
```
Comment (iluminat23) [inline customize_image.sh:68]: This comment just tells what the code does, but not why.

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: customize_image.sh (hunk 4 above)
Comment (pavlic244) [reply customize_image.sh]: I've amended the last commit , I'm ready for feedback ;-)

File: debs-to-remove [hunk 5]
```diff
@@ -10,8 +10,11 @@ dillo
 libfltk1.3
 netsurf-gtk
 netsurf-common
+libreoffice-help-en-gb
```
Comment (iluminat23) [inline debs-to-remove:13]: This would be better in a separate commit

File: debs-to-remove [hunk 6]
```diff
@@ -10,8 +10,11 @@ dillo
...
 
 # Games
+mu-editor
+code-the-classics
```
Comment (iluminat23) [inline debs-to-remove:17]: Also this

**Review by iluminat23 — CHANGES_REQUESTED** [review]

**Comment by iluminat23** [comment]
@pavlic244 The lines in the commit message should not exceed 72 characters: https://chris.beams.io/posts/git-commit/#wrap-72 I will merge it but keep this in mind.

**Review by iluminat23 — APPROVED** [review]
